packages/platform-server/src/transfer_state.ts

```diff
@@ -55,6 +87,8 @@ function serializeTransferStateFactory(doc: Document, appId: string, transferSto
       return;
     }
 
+    typeof ngDevMode !== 'undefined' && ngDevMode && warnIfStateTransferHappened(injector);
```

Comment:
nit: I'd vote to have an explicit `if` here, which makes the code more readable (in my opinion) vs having a need to scan the entire expression to get to the function call at the end:
```suggestion
    if (typeof ngDevMode !== 'undefined' && ngDevMode) {
      warnIfStateTransferHappened(injector);
    }
```